superset/viz.py

```diff
@@ -518,6 +522,19 @@ def data(self):
 
     def get_csv(self):
         df = self.get_df()
+        if df is None:
+            df = pd.DataFrame()
+            include_index = not isinstance(df.index, pd.RangeIndex)
+            return df.to_csv(index=include_index, **config.get('CSV_EXPORT'))
+        parsed_json = json.dumps(self.datasource.data, sort_keys=True)
```

Comment:
`self.datasource.data` is this full data to be exported?

Reply:
lets see if we can avoid json_dump and use python object to for extracting information.
Also i think verbose_map is available in `self.datasources.verbose_map`